bin/zowe-configure-component.sh

```diff
@@ -0,0 +1,224 @@
+#
+# Copyright Contributors to the Zowe Project. 2020
+#######################################################################
+
+#######################################################################
+# Configure Zowe Component
+#
+# This script will configure a component for Zowe instance. The component to
+# be installed can be a pax, zip or tar package, or a directory.
+#
+# Note: this script works better with NODE_HOME. But for backward compatible
+#       purpose, NODE_HOME is not mandatory.
+#
+# Command line options:
+# -c|--component-name required. component name.
+# -i|--instance-dir   required. path to Zowe instance directory.
+# -d|--target-dir     optional. directory where the component is installed.
+#                     For Zowe core component, default value is
+#                     ${ZOWE_ROOT_DIR}/components. For Zowe extensions,
+#                     the script will check ZWE_EXTENSION_DIR if possible.
+#                     Otherwise will fall back to ${DEFAULT_TARGET_DIR}.
+# -k|--core           optional boolean. Whether this component is bundled
+#                     into Zowe package.
+# -l|--logs-dir        optional. path to logs directory.
+# -f|--log-file       optional. write log to the file specified.
+#######################################################################
+
+#######################################################################
+# Constants
+DEFAULT_TARGET_DIR=/global/zowe/extensions
+
+#######################################################################
+# Prepare shell environment
+if [ -z "${ZOWE_ROOT_DIR}" ]; then
+  export ZOWE_ROOT_DIR=$(cd $(dirname $0)/../;pwd)
+fi
+
+. ${ZOWE_ROOT_DIR}/bin/internal/zowe-set-env.sh
+
+. ${ZOWE_ROOT_DIR}/bin/utils/utils.sh
+
+# node is required for read_component_manifest
+if [ -n "${NODE_HOME}" ]; then
+  ensure_node_is_on_path
+fi
+
+#######################################################################
+# Functions
+error_handler(){
+    print_error_message "$1"
+    exit 1
+}
+
+prepare_log_file() {
+    if [ -z "${LOG_FILE}" ]; then
+        set_install_log_directory "${LOG_DIRECTORY}"
+        validate_log_file_not_in_root_dir "${LOG_DIRECTORY}" "${ZOWE_ROOT_DIR}"
+        set_install_log_file "zowe-configure-component"
+    else
+        set_install_log_file_from_full_path "${LOG_FILE}"
+        validate_log_file_not_in_root_dir "${LOG_FILE}" "${ZOWE_ROOT_DIR}"
+    fi
+}
+
+enable_component(){
+    commands_start=$(read_component_manifest "${component_path}" ".commands.start" 2>/dev/null)
+    if [ "${commands_start}" != "null" ] && [ -n "${commands_start}" ] && [ "${IS_ZOWE_CORE}" = "false" ]; then
+        # we only enable if extension has commands.start defined in manifest
+        log_message "- enable ${COMPONENT_NAME} for instance ${INSTANCE_DIR}"
+        # append to EXTERNAL_COMPONENTS
+        update_zowe_instance_variable "EXTERNAL_COMPONENTS" "${COMPONENT_NAME}" "true"
+    fi
+}
+
+install_desktop_plugin(){
```

Comment:
all of the terminology around "desktop" is misleading since app framework plugins can be ZSS plugins that have no visibility in the desktop, if the developer chooses. It could merely be a rest api plugin, with no visuals.